testsupport/user_assertions.go

```diff
@@ -0,0 +1,149 @@
+		wait.UntilUserAccountHasSpec(ExpectedUserAccount(userSignup.Name, tier, templateRefs)),
+		wait.UntilUserAccountMatchesMur(hostAwait))
+	require.NoError(t, err)
+	require.NotNil(t, userAccount)
+
+	// Verify provisioned User
+	_, err = memberAwait.WaitForUser(userAccount.Name)
+	assert.NoError(t, err)
+
+	// Verify provisioned Identity
+	_, err = memberAwait.WaitForIdentity(ToIdentityName(userAccount.Spec.UserID))
+	assert.NoError(t, err)
+
+	tiers.VerifyNsTemplateSet(t, awaitility, userAccount, tier)
+
+	// Get member cluster to verify that it was used to provision user accounts
+	memberCluster, ok, err := hostAwait.GetToolchainCluster(cluster.Member, nil)
+	require.NoError(t, err)
+	require.True(t, ok)
+
+	// Then finally check again the MasterUserRecord with the expected (embedded) UserAccount status, on top of the other criteria
+	expectedEmbeddedUaStatus := toolchainv1alpha1.UserAccountStatusEmbedded{
+		Cluster: toolchainv1alpha1.Cluster{
+			Name:        mur.Spec.UserAccounts[0].TargetCluster,
+			APIEndpoint: memberCluster.Spec.APIEndpoint,
+			ConsoleURL:  ExpectedConsoleURL(t, memberAwait, memberCluster),
+		},
+		UserAccountStatus: userAccount.Status,
+	}
+	_, err = hostAwait.WaitForMasterUserRecord(mur.Name,
+		wait.UntilMasterUserRecordHasConditions(Provisioned(), ProvisionedNotificationCRCreated()),
+		wait.UntilMasterUserRecordHasUserAccountStatuses(expectedEmbeddedUaStatus))
+	assert.NoError(t, err)
+}
+
+func ExpectedUserAccount(userID string, tier string, templateRefs tiers.TemplateRefs) v1alpha1.UserAccountSpec {
+	namespaces := make([]toolchainv1alpha1.NSTemplateSetNamespace, 0, len(templateRefs.Namespaces))
+	for _, ref := range templateRefs.Namespaces {
+		namespaces = append(namespaces, toolchainv1alpha1.NSTemplateSetNamespace{
+			Template:    "", // must be empty
+			TemplateRef: ref,
+		})
+	}
+	var clusterResources *toolchainv1alpha1.NSTemplateSetClusterResources
+	if templateRefs.ClusterResources != nil {
+		clusterResources = &toolchainv1alpha1.NSTemplateSetClusterResources{
+			TemplateRef: tier + "-" + "clusterresources" + "-" + *templateRefs.ClusterResources,
+		}
+	}
+	return v1alpha1.UserAccountSpec{
+		UserID:   userID,
+		Disabled: false,
+		UserAccountSpecBase: toolchainv1alpha1.UserAccountSpecBase{
+			NSLimit: "default",
+			NSTemplateSet: toolchainv1alpha1.NSTemplateSetSpec{
+				TierName:         tier,
+				Namespaces:       namespaces,
+				ClusterResources: clusterResources,
+			},
+		},
+	}
+}
+
+func VerifyMemberStatus(t *testing.T, memberAwait *wait.MemberAwaitility) {
+	_, err := memberAwait.WaitForMemberStatus(wait.UntilMemberStatusHasConditions(ToolchainStatusReady()))
+	require.NoError(t, err, "failed while waiting for MemberStatus")
+}
+
+func VerifyToolchainStatus(t *testing.T, hostAwait *wait.HostAwaitility) {
+	_, err := hostAwait.WaitForToolchainStatus(wait.UntilToolchainStatusHasConditions(ToolchainStatusReady()))
+	require.NoError(t, err, "failed while waiting for ToolchainStatus")
+}
```

Comment:
You have put it into the the `user_assertion.go` file, but this is not related to user management verifications

Reply:
ah right, let me move them in a `toolchain_status.go` file (or something similar)

Reply:
done in 4d1424c: funcs moved into `testsupport/toolchain_status_assertions.go`